src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -94,18 +335,99 @@ export class HuiCardPicker extends LitElement {
     });
   }
 
-  private async _cardPicked(ev: Event): Promise<void> {
-    const type = (ev.currentTarget! as CardPickTarget).type;
+  private _cardPicked(ev: Event): void {
+    const config: LovelaceCardConfig = (ev.currentTarget! as CardPickTarget)
+      .config;
+
+    fireEvent(this, "config-changed", { config });
+  }
+
+  private async _getCardConfig(
+    cardConfig: CardPickerConfig
+  ): Promise<LovelaceCardConfig | undefined> {
+    if (!this.hass) {
+      return undefined;
+    }
 
-    const elClass = await getCardElementClass(type);
-    let config: LovelaceCardConfig = { type };
+    const elClass = await getCardElementClass(
+      cardConfig.lovelaceCardConfig.type
+    );
+    let config: LovelaceCardConfig = cardConfig.lovelaceCardConfig;
 
     if (elClass && elClass.getStubConfig) {
-      const cardConfig = elClass.getStubConfig(this.hass!);
-      config = { ...config, ...cardConfig };
+      const stubConfig = elClass.getStubConfig(this.hass!);
+      config = { ...stubConfig, ...config };
     }
 
-    fireEvent(this, "config-changed", { config });
+    return config;
+  }
+
+  private _getCardEntities(cardConfig: CardPickerConfig): string[] {
+    let entityIds: string[] =
+      this.entities || computeUnusedEntities(this.hass!, this.lovelace!);
```

Comment:
Use entities passed in first

Reply:
If there are `entities` passed in, we should not add others.

Reply:
If `entities` is passed in, we should only show cards that are compatible with that entity. Grey the rest out.

So if `entities` has only a light, grey out the alarm panel. (might be for a follow up PR)